internal/common/common.go

```diff
@@ -92,6 +93,13 @@ func NewProvenanceIR(binarySHA256Digest string, options ...func(p *ProvenanceIR)
 	return provenance
 }
 
+// WithBinaryName adds a binary name when creating a new ProvenanceIR.
+func WithBinaryName(binaryName string) func(p *ProvenanceIR) {
```

Comment:
I am not sure if we want to verify this, but I could be wrong. Was there anything about this when you did the requirements analysis?

Reply:
No, not at all, removed!